docs/release/handbook.md

```diff
@@ -0,0 +1,292 @@
+# Kubeflow Release Handbook
+
+## Executive Summary
+
+The purpose of this document is to define the roles and processes for the Kubeflow community to release Kubeflow. This document will attempt to coordinate releasing between the five major components and working groups in Kubeflow, as well as defining best practices, roles, and processes to make releasing Kubeflow easier for the community with each iteration.
+
+## People and Roles
+
+### Release Manager
+The Release Manager will be responsible for coordinating the release and taking ultimate accountability for all release tasks to be completed on time.
+
+**Prerequisites:**
+1. Be a contributor for at least 2 years
+2. Have shadowed in a previous release or be a Release Team member in the past
+3. Contributions in at least 2 Working Groups
```

Comment:
This requirement seems arbitrary, why is it nessesary?

Reply:
Furthermore, there may be people who are not involved in any component development working groups, but would make a good leader/member.

As is the case for SIG-Release in K8S.

Reply:
This is not arbitrary. We cannot expect a release manager to be respected, being able to have a good interaction with and making asks to other WG leaders, if this person hasn't proven they can work with WG leads smoothly in the past, by contributing to the WGs. My initial thought was to make this 3 WGs, but I relaxed this to make sure the barrier is not extremely high.

But I agree that we can completely lift this requirements for the team members, to allow more people to get involved into the process.

Reply:
what does contribution mean here? I agree with core idea of @kimwnasptd  https://github.com/kubeflow/manifests/pull/1907#discussion_r658583396. Release process/testing is one important step which ensures the final quality. While emphasising the need for bringing in more people, it is equally important for them to understand our release infra, various components and  dependencies between them. I feel, being part of a WG and driving release process of that WG for 2-3 Kubeflow releases is good entry point.

Reply:
> what does contribution mean here?

I was thinking about PRs, since this expects users to actually be involved with the code base and review/release process of WGs